pkg/zson/record.go

```diff
@@ -157,6 +173,34 @@ func (r *Record) Strings() ([]string, error) {
 	return ss, nil
 }
 
+// TypeCheck checks that the value coding in Raw is structurally consistent
+// with this value's descriptor.  It does not check that the actual leaf
+// values when parsed are type compatible with the leaf types.
+func (r *Record) TypeCheck() bool {
+	return checkRecord(r.Descriptor.Type, r.Raw)
+}
+
+func checkRecord(typ *zeek.TypeRecord, body zval.Encoding) bool {
+	it := zval.Iter(body)
+	for _, col := range typ.Columns {
+		body, container, err := it.Next()
+		if err != nil {
+			return false
+		}
+		switch v := col.Type.(type) {
+		case *zeek.TypeRecord:
+			return checkRecord(v, body)
```

Comment:
`return checkRecord(v, body)` should be preceded by 
```go
			if !container {
				return false
			}
``` 
in order to catch simple values where a record value is expected.

Reply:
yes